docs/component-libraries.md

```diff
@@ -0,0 +1,15 @@
+# Component Libraries
+
+If you find yourself asking "which `<Button>` component should I use when working with the Calypso repo?", then this is the document for you!
+
+Admittedly, our components are quite disorganized and it’s not clear what should be used, and when. There are multiple versions of (allegedly) the same component, no design system, and no coherent design language to use in order to ensure we’re building a consistent user experience. Over the years, we’ve had different representations of a components team and initiatives to unify and use the same design language everywhere, but none of them really succeeded. 
+
+When figuring out which `<Button>` component to use, here are some guidelines you could consider:
+
+* [`@wordpress/components`](https://wordpress.github.io/gutenberg) can be used for interfaces in the WordPress.org context (e.g. Block Editor).
+* [`@automattic/components`](https://wpcalypso.wordpress.com/devdocs/design) or `client/components` can be used for interfaces in the WordPress.com context.
```

Comment:
FWIW the [UI Components](https://wpcalypso.wordpress.com/devdocs/design) page will render components from both `client/components` and `@automattic/components`. We might want to update the sentence to link both to there somehow.

Reply:
866a3c7fef